### PR TITLE
Cross margin clean up and gas cost refactor

### PR DESCRIPTION
**Pull request by avclarke**
Add a new hook for `useEstimateGasCost` which returns two functions:

`estimateSnxTxGasCost(synthetixTx)`

This takes a param of synthetix tx and returns a Wei amount

`estimateEthersContractTxCost(contract, methodName, params)`

This takes params of ethers js contract, method name and method params and returns promise containing a Wei amount.

Also added a hook for useSUSDBalance so we don't need to keep repeating the same logic to filter out susd from balances.

### CODE REVIEW

File: sections/futures/PositionCard/ClosePositionModalCrossMargin.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+export default function ClosePositionModalCrossMargin({ onDismiss }: Props) {
+	const { t } = useTranslation();
+	const { handleRefetch } = useRefetchContext();
+	const { crossMarginAccountContract } = useCrossMarginAccountContracts();
+	const { monitorTransaction } = TransactionNotifier.useContainer();
+	const { resetTradeState } = useFuturesContext();
+	const { estimateEthersContractTxCost } = useEstimateGasCost();
+
+	const [crossMarginGasFee, setCrossMarginGasFee] = useState<Wei>(wei(0));
```
Comment (koredefashokun) [inline sections/futures/PositionCard/ClosePositionModalCrossMargin.tsx:32]: I think `useState` can infer the type from the initial value in this case.

File: sections/futures/PositionCard/ClosePositionModalIsolatedMargin.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	const currencyKey = useRecoilValue(currentMarketState);
+	const position = useRecoilValue(positionState);
+	const positionDetails = position?.position;
+	const selectedAccountType = useRecoilValue(futuresAccountTypeState);
+
+	const gasPrice = ethGasPriceQuery.data?.[gasSpeed] ?? null;
+
+	const closeTxn = useSynthetixTxn(
+		`FuturesMarket${currencyKey?.[0] === 's' ? currencyKey.substring(1) : currencyKey}`,
```
Comment (koredefashokun) [inline sections/futures/PositionCard/ClosePositionModalIsolatedMargin.tsx:36]: I think the `getDisplayAsset` function in `utils/futures.ts` can do this.

File: sections/futures/TradeCrossMargin/DepositWithdrawCrossMargin.tsx [hunk 3]
```diff
@@ -57,13 +59,27 @@ export default function DepositWithdrawCrossMargin({
...
 	const susdBal = transferType === 0 ? balances?.susdWalletBalance || zeroBN : freeMargin;
+	const { estimateEthersContractTxCost } = useEstimateGasCost();
+
+	useEffect(() => {
+		if (!crossMarginAccountContract) return;
+		const estimateGas = async () => {
+			if (!amount) return;
+			const method = transferType === 0 ? 'deposit' : 'withdraw';
+			const fee = await estimateEthersContractTxCost(crossMarginAccountContract, method, [
+				wei(amount || 0).toBN(),
```
Comment (koredefashokun) [inline sections/futures/TradeCrossMargin/DepositWithdrawCrossMargin.tsx:70]: Looks like `amount` can not be falsy here, because of the check on line 67.

**Review by koredefashokun — COMMENTED** [review]
The changes look good. Just left a couple of comments on style/lint things.

File: hooks/useEstimateGasCost.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	const { useEthGasPriceQuery, useExchangeRatesQuery } = useSynthetixQueries();
+	const ethGasPriceQuery = useEthGasPriceQuery();
+	const exchangeRatesQuery = useExchangeRatesQuery();
+	const getL1Fee = useGetL1SecurityFee();
+
+	const gasPrice = ethGasPriceQuery.data != null ? ethGasPriceQuery.data[gasSpeed] : null;
+
+	const exchangeRates = useMemo(
+		() => (exchangeRatesQuery.isSuccess ? exchangeRatesQuery.data ?? null : null),
```
[the remark this answers is not in the document]
Comment (avclarke) [reply hooks/useEstimateGasCost.ts:26]: Yea good call, was a copy paste there

File: sections/futures/PositionCard/ClosePositionModalIsolatedMargin.tsx (hunk 2 above)
Comment (avclarke) [reply sections/futures/PositionCard/ClosePositionModalIsolatedMargin.tsx:36]: nice, will update

**Review by koredefashokun — APPROVED** [review]

**Pull request merged**